### DISCUSSION TITLE
Make failure

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#28724] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I am trying to compile my application and I got this feedback. Is it an error? Was the compilation successful?
```
(python-stack) [emmanuel324@tinkercliffs1 belson]$ make -j8
Creating Unity /home/emmanuel324/projects/moose/modules/solid_mechanics/build/unity_src/materials_Unity.C
Creating Unity /home/emmanuel324/projects/belson/build/unity_src/materials_Unity.C
Compiling C++ (in opt mode) /home/emmanuel324/projects/moose/modules/solid_mechanics/build/unity_src/materials_Unity.C...
Compiling C++ (in opt mode) /home/emmanuel324/projects/belson/build/unity_src/materials_Unity.C...
Linking Library /home/emmanuel324/projects/moose/modules/solid_mechanics/lib/libsolid_mechanics-opt.la...
Linking Library /home/emmanuel324/projects/moose/modules/contact/lib/libcontact-opt.la...
Linking Library /home/emmanuel324/projects/moose/modules/fsi/lib/libfsi-opt.la...
Linking Library /home/emmanuel324/projects/moose/modules/peridynamics/lib/libperidynamics-opt.la...
Linking Library /home/emmanuel324/projects/moose/modules/phase_field/lib/libphase_field-opt.la...
Linking Library /home/emmanuel324/projects/moose/modules/porous_flow/lib/libporous_flow-opt.la...
Linking Library /home/emmanuel324/projects/moose/modules/xfem/lib/libxfem-opt.la...
Linking Library /home/emmanuel324/projects/moose/modules/combined/lib/libcombined-opt.la...
Linking Library /home/emmanuel324/projects/moose/modules/module_loader/lib/libmodule_loader_with_cr_em_eps_fp_fet_gc_ls_ray_rdg_rct_rich_st_ht_sp_sm_ns_con_fsi_misc_opt_pd_pf_pflow_th_st_xfe
m_comb-opt.la...
Linking Library /home/emmanuel324/projects/belson/lib/libbelson-opt.la...
Linking Test Library /home/emmanuel324/projects/belson/test/lib/libbelson_test-opt.la...
Linking Executable /home/emmanuel324/projects/belson/belson-opt...
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: skipping incompatible /lib/libutil.so when searching for -lutil
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: skipping incompatible /lib/libpthread.so when searching for -lpthread
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: skipping incompatible /lib/librt.so when searching for -lrt
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: skipping incompatible /lib/libdl.so when searching for -ldl
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: skipping incompatible /lib/libm.so when searching for -lm
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: skipping incompatible /lib/libpthread.so when searching for -lpthread
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: skipping incompatible /lib/libc.so when searching for -lc
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: /home/emmanuel324/projects/belson/lib/libbelson-opt.so: undefined reference to `InputParameters validParams<Or
derParameterFunctionMaterial>()'
/apps/easybuild/software/tinkercliffs-rome/binutils/2.40-GCCcore-13.2.0/bin/ld: /home/emmanuel324/projects/belson/lib/libbelson-opt.so: undefined reference to `InputParameters validParams<Ke
rnel>()'
collect2: error: ld returned 1 exit status
make: *** [/home/emmanuel324/projects/belson/belson-opt] Error 1
(python-stack) [emmanuel324@tinkercliffs1 belson]$ 
```

**Comment by GiudGiud** [comment]
This syntax for parameters has been deprecated for a while. 

You need to update those two objects to the new syntax, git clean and recompile

**Reply by Emmanuel-324** [reply]
Can you give a little bit of detail on what I  need to do resolve this. I just added two source codes and header files in my application, that caused this issue.

**Reply by GiudGiud** [reply, marked as answer]
see this announcement
https://mooseframework.inl.gov/docs/PRs/28672/site/newsletter/2021/2021_11.html#legacy-input-parameter-deprecation

**Reply by Emmanuel-324** [reply]
Thanks, issues resolved!

**Reply by Emmanuel-324** [reply]
My app compiled successfully but I still get the error with my script
```
A 'BarrierFunctionMaterial_abs' is not a registered object.check_inp(moose_srv)
If you are trying to find this object in a dynamically loaded library, make sure thatcheck_inp(moose_srv)
the library can be found either in your "Problem/library_path" parameter or in thecheck_inp(moose_srv)
MOOSE_LIBRARY_PATH environment variable
```
How do I fix it?

**Reply by GiudGiud** [reply]
how did you register your material? can you paste the beginning of the source file for this material?

**Reply by Emmanuel-324** [reply]
```
//* This file is part of the MOOSE framework
//* https://www.mooseframework.org
//*
//* All rights reserved, see COPYRIGHT for full restrictions
//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
//*
//* Licensed under LGPL 2.1, please see LICENSE for details
//* https://www.gnu.org/licenses/lgpl-2.1.html

#include "BarrierFunctionMaterial_abs.h"

registerMooseObject("belson", BarrierFunctionMaterial_abs);


InputParameters
BarrierFunctionMaterial_abs::validParams()
{
  InputParameters params = OrderParameterFunctionMaterial::validParams();
  params.addClassDescription("Helper material to provide g(eta) and its derivative in a "
                             "polynomial.\nSIMPLE: eta^2*(1-eta)^2\nLOW: eta*(1-eta)"
                             "\nHIGH: eta^2*(1-eta^2)^2");
  MooseEnum g_order("SIMPLE=0 LOW HIGH", "SIMPLE");
  params.addParam<MooseEnum>("g_order", g_order, "Polynomial order of the barrier function g(eta)");
  params.addParam<bool>("well_only",
                        false,
                        "Make the g zero in [0:1] so it only contributes to "
                        "enforcing the eta range and not to the phase "
                        "transformation berrier.");
  params.set<std::string>("function_name") = std::string("g");
  return params;
}
```

**Reply by GiudGiud** [reply]
it should probably be 
```
registerMooseObject("belsonApp", BarrierFunctionMaterial_abs);
```
if that does not work, try
```
registerMooseObject("MooseApp", BarrierFunctionMaterial_abs);
```

**Reply by Emmanuel-324** [reply]
Thanks the first suggestion worked

**Reply by ColinMG2** [reply]
Hello! I have run into the same error when trying to run make for the solid mechanics module. I tried clicking on the link you posted but I get a 404 not found error. How can I resolve this issue. My output is similar: 
(moose) colinmoose@ColinG14:~/projects/moose/modules/solid_mechanics$ make -j 16
Linking Executable /home/colinmoose/projects/moose/modules/solid_mechanics/solid_mechanics-opt...
/home/colinmoose/miniforge/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/colinmoose/projects/moose/modules/solid_mechanics/lib/libsolid_mechanics-opt.so: undefined reference to `non-virtual thunk to ExplicitTimeIntegrator::meshChanged()'
collect2: error: ld returned 1 exit status
make: *** [/home/colinmoose/projects/moose/framework/app.mk:466: /home/colinmoose/projects/moose/modules/solid_mechanics/solid_mechanics-opt] Error 1

**Reply by GiudGiud** [reply]
Weird error. Is this a fresh build? It could be some leftover compiles objects from before that object was modified (in which case you would to save your work using git commit then use git clean -Xfd in the moose/ folder to delete them, then compile again

**Reply by ColinMG2** [reply]
I recently updated moose and conda and yes I have never run the make function for the solid mechanics module. I can try running the git commands you suggested.

**Reply by ColinMG2** [reply]
I ran git commit and it said that there was nothing to commit: 
(moose) colinmoose@ColinG14:~/projects/moose$ git commit
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

**Reply by ColinMG2** [reply]
I wiped everything clean in the cd projects/moose folder using git clean -Xfd , however, when I tried to recompile by running make -j 16 again I got this error: 
x86_64-conda-linux-gnu-c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make: *** [/home/colinmoose/projects/moose/framework/build.mk:150: /home/colinmoose/projects/moose/framework/build/unity_src/kernels_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....

**Reply by GiudGiud** [reply]
This error means you ran out of memory. You can try to reduce the number of processes when building

**Reply by GiudGiud** [reply]
So make -j10 first? Then maybe less if it persists

**Reply by ColinMG2** [reply]
Okay I see. When I looked at the Moose Training Workshop slides it said when running make you should use the number of cores on your computer processor. -j 10 seemed to work and I ran tests and they all passed. Thanks!
